### PR TITLE
ARROW-13734: [Format] Clarify allowed values for time types

### PR DESCRIPTION
**Pull request by pitrou**

### CODE REVIEW

File: format/Schema.fbs [hunk 1]
```diff
@@ -206,9 +206,20 @@ table Date {
...
+/// microseconds or nanoseconds.
+///
+/// The integer `bitWidth` depends on the `unit` and must be one of the following:
+/// * SECOND and MILLISECOND: 32 bits
+/// * MICROSECOND and NANOSECOND: 64 bits
+///
+/// The allowed values are between 0 (inclusive) and 86400 (exclusive), adjusted
+/// for the time unit (for example, up to 86400000 exclusive for the MILLISECOND
+/// unit).
+/// This definition doesn't allow for leap seconds.  If converting data from a
```
Comment (jorgecarleitao) [inline format/Schema.fbs:220]:
```suggestion
/// This definition doesn't allow for leap seconds. If converting data from a
```

File: format/Schema.fbs [hunk 2]
```diff
@@ -206,9 +206,20 @@ table Date {
...
-/// - MICROSECOND and NANOSECOND: 64 bits
+/// Time is either a 32-bit or 64-bit integer type representing an elapsed time
+/// since midnight, stored in either of four units: seconds, milliseconds,
+/// microseconds or nanoseconds.
+///
+/// The integer `bitWidth` depends on the `unit` and must be one of the following:
+/// * SECOND and MILLISECOND: 32 bits
+/// * MICROSECOND and NANOSECOND: 64 bits
+///
+/// The allowed values are between 0 (inclusive) and 86400 (exclusive), adjusted
```
Comment (jorgecarleitao) [inline format/Schema.fbs:217]:
```suggestion
/// The allowed values are between 0 (inclusive) and 86400 (=24*60*60) seconds (exclusive), adjusted
```

File: format/Schema.fbs [hunk 3]
```diff
@@ -194,8 +194,8 @@ enum DateUnit: short {
   MILLISECOND
 }
 
-/// Date is either a 32-bit or 64-bit type representing elapsed time since UNIX
-/// epoch (1970-01-01), stored in either of two units:
+/// Date is either a 32-bit or 64-bit integer type representing an elapsed time
```
Comment (jorgecarleitao) [inline format/Schema.fbs:197]:
```suggestion
/// Date is either a 32-bit or 64-bit signed integer type representing an elapsed time
```

File: format/Schema.fbs [hunk 4]
```diff
@@ -206,9 +206,20 @@ table Date {
 
 enum TimeUnit: short { SECOND, MILLISECOND, MICROSECOND, NANOSECOND }
 
-/// Time type. The physical storage type depends on the unit
-/// - SECOND and MILLISECOND: 32 bits
-/// - MICROSECOND and NANOSECOND: 64 bits
+/// Time is either a 32-bit or 64-bit integer type representing an elapsed time
```
Comment (jorgecarleitao) [inline format/Schema.fbs:209]:
```suggestion
/// Time is either a 32-bit or 64-bit signed integer type representing an elapsed time
```

File: format/Schema.fbs [hunk 5]
```diff
@@ -206,9 +206,20 @@ table Date {
...
+///
+/// The integer `bitWidth` depends on the `unit` and must be one of the following:
+/// * SECOND and MILLISECOND: 32 bits
+/// * MICROSECOND and NANOSECOND: 64 bits
+///
+/// The allowed values are between 0 (inclusive) and 86400 (exclusive), adjusted
+/// for the time unit (for example, up to 86400000 exclusive for the MILLISECOND
+/// unit).
+/// This definition doesn't allow for leap seconds.  If converting data from a
+/// system which may produce leap seconds, it will be necessary to adjust incoming
```
Comment (jorgecarleitao) [inline format/Schema.fbs:221]: alternative wording:

> Times from systems with leap seconds must be adjusted to the leap-less format when converted to this system (for example by converting 86400 to 86399).

**Review by jorgecarleitao — APPROVED** [review]
Thanks for the write! looks great; left some suggestions.

**Review by lidavidm — APPROVED** [review]
Looks good to me, thanks for clarifying these points.

**Pull request closed without merging**